### ISSUE TITLE
[FEATURE] first_true

### ISSUE THREAD

**Issue by DenisYaroshevskiy** [jfalcou/eve#449]
Depends on: https://github.com/jfalcou/eve/issues/446

Implement a function:

```
first_true[relative_conditional_expr](logical) -> std::optional<std::ptrdiff_t>
```

For x86 can be implemented by forwarding to the `top_bits` abstraction.

**Comment by jfalcou** [comment]
I guess we also want `first_true(logical)` for completeness?

**Comment by DenisYaroshevskiy** [comment]
@jfalcou - I am not sure - what does it do?

```
first_true[logial](logical) == first_true(logical && logical)
```
?

I am not super convinced, to be honest, that `ignore` and `logical` are one in a same. We don't have `load[logical]` I believe, do we? I dunno - we of course can make it work but I am not sure it makes a lot of sence.

**Comment by jfalcou** [comment]
No just first_true(logical) that checks the first true index of a logical without any masking.

**Comment by DenisYaroshevskiy** [comment]
@jfalcou  ah yes :) sorry

**Comment by DenisYaroshevskiy** [comment]
Closed by accident

**Comment by DenisYaroshevskiy** [comment]
Can be closed, remaining are in a separate issue
